File: ecommerce/src/main/java/ecommerce/main/Model/CustomerModel.java
package ecommerce.main.Model;

import ecommerce.main.Entities.Address;
import ecommerce.main.Entities.Payment;
import lombok.Data;

@Data
public class CustomerModel {
    private Integer id;
    private String name;
    private String email;
    private String password;
    private Address address;
    private Payment payment;
}

File: ecommerce-frontend/src/app/core/models/customer.model.ts
import { Address } from "./address.model";
import { Payment } from "./payment.model";

export class CustomerSave{
    constructor(
        public id: number,
        public name: string,
        public email: string,
        public password: string, 
        public address: Address,
        public payment: Payment
    ){}
}

export class CustomerGetProfile{
    constructor(
        public email: string, 
        public id: number
    ){}
}

export class CustomerLogin{
    constructor(
        public email: string,
        public password: string
    ){}
}

export class Customer{
    public password: string;
    public address: Address;
    public payment: Payment;
    
    constructor(
        public id: number,
        public name: string,         
        public email: string,        
        private _token: string,
        private _tokenExpirationDate: Date        
    ){}

    get token(){
        if(!this._tokenExpirationDate || new Date() > this._tokenExpirationDate)
            return null

        return this._token;
    }
}